numpy/random/bit_generator.pyx

```diff
@@ -0,0 +1,496 @@
+        if self.program_entropy is None:
+            # We *could* make `_coerce_to_uint32_array(None)` handle this case,
+            # but that would make it easier to misuse, a la
+            # `_coerce_to_uint32_array([None, 12345])`
+            program_entropy = np.array([], dtype=np.uint32)
+        else:
+            program_entropy = _coerce_to_uint32_array(self.program_entropy)
+        spawn_entropy = _coerce_to_uint32_array(self.spawn_key)
+        entropy_array = np.concatenate([run_entropy, program_entropy,
+                                        spawn_entropy])
+        return entropy_array
+
+    @np.errstate(over='ignore')
+    def generate_state(self, n_words, dtype=np.uint32):
+        """ Return the requested number of words for PRNG seeding.
+
+        A BitGenerator should call this method in its constructor with
+        an appropriate `n_words` parameter to properly seed itself.
+
+        Parameters
+        ----------
+        n_words : int
+        dtype : np.uint32 or np.uint64, optional
+            The size of each word. This should only be either `uint32` or
+            `uint64`. Strings (`'uint32'`, `'uint64'`) are fine. Note that
+            requesting `uint64` will draw twice as many bits as `uint32` for
+            the same `n_words`. This is a convenience for `BitGenerator`s that
+            express their states as `uint64` arrays.
+        Returns
+        -------
+        state : uint32 or uint64 array, shape=(n_words,)
+        """
+        cdef uint32_t hash_const = INIT_B
+        cdef uint32_t data_val
+
+        out_dtype = np.dtype(dtype)
+        if out_dtype == np.dtype(np.uint32):
+            pass
+        elif out_dtype == np.dtype(np.uint64):
+            n_words *= 2
+        else:
+            raise ValueError("only support uint32 or uint64")
+        state = np.zeros(n_words, dtype=np.uint32)
+        src_cycle = cycle(self.pool)
+        for i_dst in range(n_words):
+            data_val = next(src_cycle)
+            data_val ^= hash_const
+            hash_const *= MULT_B
+            data_val *= hash_const
+            data_val ^= data_val >> XSHIFT
+            state[i_dst] = data_val
+        if out_dtype == np.dtype(np.uint64):
+            state = state.view(np.uint64)
+        return state
+
+    def spawn(self, n_children):
+        """ Spawn a number of child `SeedSequence`s by extending the
+        `spawn_key`.
+
+        Parameters
+        ----------
+        n_children : int
+        Returns
+        -------
+        seqs : list of `SeedSequence`s
+        """
+        seqs = []
+        for i in range(self.n_children_spawned,
+                       self.n_children_spawned + n_children):
+            seqs.append(SeedSequence(
+                self.entropy,
+                program_entropy=self.program_entropy,
+                spawn_key=self.spawn_key + (i,),
+                pool_size=self.pool_size,
+            ))
+        return seqs
```

Comment:
`self.n_children_spawned` needs to be incremented.

Reply:
good catch, fixing